client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -0,0 +1,217 @@
+import { DraftailEditor, ToolbarButton, createEditorStateFromRaw, serialiseEditorStateToRaw } from 'draftail';
+import { EditorState, Modifier, RichUtils, SelectionState } from 'draft-js';
+import { filterInlineStyles } from "draftjs-filters";
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { useSelector, shallowEqual } from 'react-redux';
+
+import { STRINGS } from '../../../config/wagtailConfig';
+import Icon from '../../Icon/Icon';
+
+const COMMENT_STYLE_IDENTIFIER = 'COMMENT-'
+
+function usePrevious(value) {
+  const ref = useRef();
+  useEffect(() => {
+    ref.current = value;
+  }, [value]); // Only re-run if value changes
+  return ref.current;
+}
+
+class DraftailInlineAnnotation {
+    constructor(field) {
+      this.field = field;
+      this.inlineRefs = new Set()
+    }
+    addRef(ref) {
+      this.inlineRefs.add(ref);
+    }
+    removeRef(ref) {
+      this.inlineRefs.delete(ref);
+    }
+    getDesiredPosition() {
+      const nodeTops = Array.from(this.inlineRefs).map(ref => ref.current).filter(node => node).map(node => node.getBoundingClientRect().top)
+      if (nodeTops.length > 0) {
+        return nodeTops.reduce((a,b) => a + b, 0)/nodeTops.length + document.documentElement.scrollTop
```

Comment:
I think this would warrant either a comment, or a variable name to explain what this is? I’m surprised we’d want an average here.

Reply:
I'll add a comment - you're right, it's not necessarily that clear. Why are you surprised we'd want an average, if you don't mind me asking? To me if you've got a few different highlights, it makes sense for the comment to try and float vertically in the middle of the set.

Reply:
This is really just the usual "mean vs median" question in statistics. Looking at the particular consequences here, both suffer from the problem that you can have inline annotations quite far apart from one-another in a long field.

I’m not sure I fully understand the comment positioning logic so I’ll only state what I think we want:

- If a comment is attached to multiple inline annotations, I think it should either be displayed next to the first of those annotations, or next to the "median" annotation. Currently using the average means that comments can be placed alongside seemingly unrelated text.
- Comments should always be displayed in the viewport when focused. 

Here is an extreme case for reference:

![comments-position](https://user-images.githubusercontent.com/877585/110477926-a9ce6100-80db-11eb-83d2-32220f7a5412.gif)

Reply:
Addressed now